### PR TITLE
Migrate from optparse to argparse

### LINKED ISSUE #7158: Update code using optparse to argparse
optparse is deprecated since py2.7. At least the reqmgr client scripts use it.

### PR DESCRIPTION
**Pull request by amaltaro**
Fixes #7158 
Resuming the work done by Alex on #8202

Very annoying PR, some files can likely be deleted (etc/harvestingInjector).
I also had to replace `raw_input` by `input`, otherwise python3 compatibility tests would start failing. Notice that with that change, we can no longer type only `yes` in the prompt, but actually `"yes"` has to be typed...

### CODE REVIEW

**Comment by amaltaro** [comment]
test this please

File: bin/wmagent-delete-couchdb-workflow [hunk 1]
```diff
@@ -26,43 +30,43 @@ def deleteWorkflowFromCouch(workflowName, jobsdatabase, fwjrdatabase):
...
+        fwjrdatabase.delete_doc(id=id, rev=rev)
 
     return len(jobs), len(fjobs)
 
-if __name__ == "__main__":
 
-    answer = raw_input("Warning: This is irreversible. Are you sure to delete workflows (yes, no)? ")
+if __name__ == "__main__":
+    print("Warning: This is irreversible. ")
+    answer = input("Are you sure to delete workflows (yes, no, type it between quotes)? ")
```
Comment (ticoann) [inline bin/wmagent-delete-couchdb-workflow]: I think you can explicitly say 'yes', 'no'  like above instruction
```
Type 'Y' or 'N' (type quotes too!)
```

File: bin/wmagent-unregister-wmstats [hunk 2]
```diff
@@ -1,40 +1,42 @@
...
+    (options, args) = parser.parse_known_args()
     if not args:
         agentUrl = ("%s" % wmagentConfig.Agent.hostName)
     else:
         agentUrl = args[0]
-    
-    answer = raw_input("Warning: This is remove agent info record for %s from wmstats (yes, no)?" % agentUrl)
+
+    msg = "Are you sure you want to remove agent info record for %s from wmstats (yes, no, between quotes please)? "
+    answer = input(msg % agentUrl)
```
Comment (ticoann) [inline bin/wmagent-unregister-wmstats]: Same as above. (When we move to python3 quotation need to be dropped, right?)

**Comment by amaltaro** [comment]
Yes, when we move to python3, `input` will have the same behavior as `raw_input`, so it won't try to evaluate the input as an expression.
I'll update the message then.

**Pull request merged**